model/perf.go

```diff
@@ -656,13 +657,90 @@ func (r *PerformanceResults) FindOutdatedRollups(ctx context.Context, name strin
 	return errors.WithStack(it.Close(ctx))
 }
 
-func GetTimeSeriesIds(ctx context.Context, env cedar.Environment) ([]TimeSeriesId, error) {
-	pipeline := []bson.M{
+func createTimeStamp(ctx context.Context, env cedar.Environment, id TimeSeriesId, field string) error {
+	_, err := env.GetDB().Collection(metricMetadataCollection).UpdateOne(ctx, id, bson.M{
+		"$currentDate": bson.M{
+			field: true,
+		},
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Unable to create timestamp")
+	}
+	return nil
+}
+
+func MarkMetricAsUpdated(ctx context.Context, env cedar.Environment, id TimeSeriesId) error {
+	err := createTimeStamp(ctx, env, id, "lastTimeSeriesUpdate")
+	if err != nil {
+		return errors.Wrapf(err, "Unable to mark metric as updated")
+	}
+	return nil
+}
+
+func MarkMetricAsAnalyzed(ctx context.Context, env cedar.Environment, id TimeSeriesId) error {
+	err := createTimeStamp(ctx, env, id, "lastChangePointDetection")
+	if err != nil {
+		return errors.Wrapf(err, "Unable to mark metric as analyzed for change points")
+	}
+	return nil
+}
+
+func GetMetricsNeedingChangePointDetection(ctx context.Context, env cedar.Environment) ([]TimeSeriesId, error) {
+	cur, err := env.GetDB().Collection(metricMetadataCollection).Find(ctx, bson.M{
+		"$expr": bson.M{
+			"$lt": []string{
+				"$lastChangePointDetection",
+				"$lastTimeSeriesUpdate",
+			},
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get metrics needing change point detection")
+	}
+	defer cur.Close(ctx)
+	var res []struct {
```

Comment:
can you declare a type in the package that has these values, with the MustHaveTag values?

Reply:
Think this has been handled.